apollo-router/src/http_service_registry.rs

```diff
@@ -27,27 +31,61 @@ impl HttpServiceRegistry {
                 .subgraphs
                 .iter()
                 .map(|(name, subgraph)| {
-                    let fetcher: Box<dyn graphql::Fetcher> = Box::new(HttpSubgraphFetcher::new(
-                        name.to_owned(),
-                        subgraph.routing_url.to_owned(),
-                    ));
-                    (name.to_string(), fetcher)
+                    let fetcher =
+                        HttpSubgraphFetcher::new(name.to_owned(), subgraph.routing_url.to_owned());
+                    let cloner: Arc<dyn NewSubgraphService> = Arc::new(move || fetcher.clone());
+                    (name.to_string(), cloner)
                 })
                 .collect(),
         }
     }
+
+    // test dynamically adding a Service
+    pub fn add_service<S>(&mut self, name: String, service: S)
+    where
+        S: Clone + SubgraphService,
```

Comment:
I'm not sure why you would need to clone it before pushing it to the map. I doubt the user would need to do something with the service after it has been added using this function :thinking:

Reply:
it is not cloned, it is moved in a closure that clones it on each call